modules/host/persist.go

```diff
@@ -1,80 +1,165 @@
 package host
 
 import (
-	"log"
 	"os"
 	"path/filepath"
 
+	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/persist"
 	"github.com/NebulousLabs/Sia/types"
 )
 
+const (
+	// logFile establishes the name of the file that gets used for logging.
+	logFile = modules.HostDir + ".log"
+)
+
+// Variables indicating the metadata header and the version of the data that
+// has been saved to disk.
 var persistMetadata = persist.Metadata{
 	Header:  "Sia Host",
-	Version: "0.4",
+	Version: "0.5",
 }
 
-type savedHost struct {
-	SpaceRemaining int64
-	FileCounter    int
-	Profit         types.Currency
-	HostSettings   modules.HostSettings
-	Obligations    []contractObligation
-	SecretKey      crypto.SecretKey
-	PublicKey      types.SiaPublicKey
+// persistence is the data that is kept when the host is restarted.
+type persistence struct {
+	// Host Context.
+	BlockHeight  types.BlockHeight
+	RecentChange modules.ConsensusChangeID
+	PublicKey    types.SiaPublicKey
+	SecretKey    crypto.SecretKey
+	Settings     modules.HostSettings
+
+	// File Management.
+	FileCounter int64
+	Obligations []contractObligation
+
+	// Statistics.
+	Profit types.Currency
 }
 
-func (h *Host) save() error {
-	sHost := savedHost{
-		SpaceRemaining: h.spaceRemaining,
-		FileCounter:    h.fileCounter,
-		Profit:         h.profit,
-		HostSettings:   h.HostSettings,
-		Obligations:    make([]contractObligation, 0, len(h.obligationsByID)),
-		SecretKey:      h.secretKey,
-		PublicKey:      h.publicKey,
-	}
+// getObligations returns a slice containing all of the contract obligations
+// currently being tracked by the host.
+func (h *Host) getObligations() []contractObligation {
+	cos := make([]contractObligation, 0, len(h.obligationsByID))
 	for _, ob := range h.obligationsByID {
-		// to avoid race conditions involving the obligation's mutex, copy it
-		// manually into a new object
```

Comment:
I don't think that it's correct, in fact it's not clear to me that you can't just call `cos = append(cos, ob)`. The obligation's mutex doesn't cover any of the data in the structure, only the actual act of doing a revision.

And if you are going to be doing revisions on the obligation, that's an indication to me that you should be passing the whole obligation by reference instead of by value.  Maybe it should be passed by reference?

I don't think there's a race condition here, though `go vet` will complain about passing a mutex by value.